src/tools/r2rdump/R2RReader.cs

```diff
@@ -77,13 +83,65 @@ public unsafe R2RReader(string filename)
                 }
 
                 DirectoryEntry r2rHeaderDirectory = peReader.PEHeaders.CorHeader.ManagedNativeHeaderDirectory;
-                int r2rHeaderOffset = r2rHeaderDirectory.RelativeVirtualAddress - textSection.VirtualAddress + textSection.PointerToRawData;
-                R2RHeader = new R2RHeader(_image, (uint)r2rHeaderDirectory.RelativeVirtualAddress, r2rHeaderOffset);
+                int r2rHeaderOffset = GetOffset(r2rHeaderDirectory.RelativeVirtualAddress, textSection);
+                R2RHeader = new R2RHeader(_image, r2rHeaderDirectory.RelativeVirtualAddress, r2rHeaderOffset);
                 if (r2rHeaderDirectory.Size != R2RHeader.Size)
                 {
                     throw new System.BadImageFormatException("The calculated size of the R2RHeader doesn't match the size saved in the ManagedNativeHeaderDirectory");
                 }
+
+                R2RSection runtimeFunctions = R2RHeader.Sections[R2RSection.SectionType.READYTORUN_SECTION_RUNTIME_FUNCTIONS];
+                int curOffset = GetOffset((int)runtimeFunctions.RelativeVirtualAddress, textSection);
+                int nNativeCode = runtimeFunctions.Size / (3*sizeof(int));
```

Comment:
This multiplication by 3 seems arbitrary without any comments. Could you please leave a comment explaining why we need that?

Reply:
It is also platform-specific. It is 2 on other platforms.

Reply:
The multiplication by 3 happens because each method has 3 RVAs (start RVA, end RVA and the RVA of the unwind info). Is that different for other platforms?

Reply:
Ah I see what Jan is saying. You can see in https://github.com/dotnet/coreclr/blob/8a375fe25d8b5f1a2fadb8afbf3360a4da67f71b/src/inc/crosscomp.h that we define the RUNTIME_FUNCTION (or T_RUNTIME_FUNCTION) for different architectures.

Reply:
Actually, Jan may be referring to https://github.com/dotnet/coreclr/blob/9c110c3b1200fdade1c85df5da58f0a952b0754e/src/pal/inc/pal.h#L2993